### PR TITLE
Don't resolve filepaths

### LINKED ISSUE #218: Don't resolve filepaths
Current `data_dir` gets resolved to an absolute path, which causes caching to fail.

The fix is to leave it alone -- nothing wrong with relative paths.

### PR DESCRIPTION
**Pull request by AllenDowney**
Current `data_dir` gets resolved to an absolute path, which causes caching to fail.

The fix is to leave it alone -- nothing wrong with relative paths.

Closes #218

### CODE REVIEW

File: zamba/models/config.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def check_files_exist_and_load(
     logger.info(
         f"Checking all {len(files_df):,} filepaths exist. Can take up to a minute for every couple thousand files."
     )
-    invalid_files = files_df[~files_df.filepath.path.exists()]
+    exists = files_df['filepath'].path.exists()
```
Comment (AllenDowney) [inline zamba/models/config.py:107]: This is an unrelated change I made to help with debugging.

File: zamba/models/config.py [hunk 2]
```diff
@@ -93,8 +93,8 @@ def check_files_exist_and_load(
     Returns:
         pd.DataFrame: DataFrame with valid and loadable videos.
     """
-    # update filepath column to prepend data_dir if filepath column is not an absolute path
-    data_dir = Path(data_dir).resolve()
+    # update filepath column to prepend data_dir
+    # NOTE: A previous version of this resolved all paths to absolute, but that breaks caching.
```
Comment (ejm714) [inline zamba/models/config.py:97]: [nit] do you think we need this note here? the only reason this breaks caching is because we provided a previously cached directory (which is not a supported user use case). the absolute path works fine if you let zamba handle the caching on the first epoch. i'm inclined to remove this comment since i think it contains inside baseball and doesn't help us in the future

File: zamba/models/config.py [hunk 3]
```diff
@@ -355,7 +356,7 @@ class TrainConfig(ZambaBaseModel):
     """
 
     labels: Union[FilePath, pd.DataFrame]
-    data_dir: DirectoryPath = Path.cwd()
+    data_dir: Optional[DirectoryPath] = ""
```
Comment (ejm714) [inline zamba/models/config.py:359]: `Optional` means DirectoryPath can be `None` (https://pydantic-docs.helpmanual.io/usage/types/). I don't think `None` actually works otherwise you'll get `str(None) / df.filepath.path`. I think we just want this to be `DirectoryPath` (as you have in the PredictConfig)

**Review by ejm714 — CHANGES_REQUESTED** [review]
One small tweak needed and one optional one, then it will be good to 🚀

File: zamba/models/config.py (hunk 2 above)
Comment (AllenDowney) [reply zamba/models/config.py:97]: Agreed

File: zamba/models/config.py (hunk 3 above)
Comment (AllenDowney) [reply zamba/models/config.py:359]: Yes.

**Review by ejm714 — APPROVED** [review]
LGMT! @AllenDowney feel free to merge once the tests finish passing
